### PR TITLE
Added autogluon.features module

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- [x] Moved FeatureMetadata to autogluon.core
- [x] Moved AbstractModel to autogluon.core
- [x] Moved FeatureGenerators to autogluon.features

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-889-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-889/3/index.html

**Comment by szha** [comment]
Job PR-889-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-889/4/index.html

**Comment by szha** [comment]
Job PR-889-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-889/5/index.html

**Comment by szha** [comment]
Job PR-889-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-889/7/index.html

**Comment by szha** [comment]
Job PR-889-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-889/8/index.html

File: Jenkinsfile [hunk 1]
```diff
@@ -24,6 +24,7 @@ stage("Unit Test") {
           pip uninstall -y autogluon.mxnet
           pip uninstall -y autogluon.extra
           pip uninstall -y autogluon.tabular
+          pip uninstall -y autogluon.features
```
Comment (sxjscience) [inline Jenkinsfile:27]: I think that it should be in `autogluon.core` should we install another `autogluon.features` module?

Comment (Innixma) [reply Jenkinsfile:27]: What do you mean? Are you saying that autogluon.core should depend on autogluon.features?

Comment (sxjscience) [reply Jenkinsfile:27]: Per-offline discussion, @Innixma mentioned that we may later expand the functionality of `autogluon.features` so it's beneficial to have a separate `autogluno.features` package.

**Review by gradientsky — APPROVED** [review]

**Review by sxjscience — APPROVED** [review]
LGTM

**Pull request merged**